src/rpc/tokens.rs

```diff
@@ -0,0 +1,159 @@
+        url.set_path("api/v1/tokens/");
+        url
+    }
+
+    /// Instantiate from a client
+    pub(crate) fn new(client: &'a SafeClient) -> Self {
+        Self {
+            filters: Default::default(),
+            client,
+        }
+    }
+
+    fn clear_decimals(&mut self) {
+        for k in Self::DECIMAL_KEYS {
+            self.filters.remove(k);
+        }
+    }
+
+    /// Filters tokens by name
+    pub fn name(mut self, name: String) -> Self {
+        self.filters.insert("name", name);
+        self
+    }
+
+    /// Filter tokens by address
+    pub fn address(mut self, address: Address) -> Self {
+        self.insert("address", address);
+        self
+    }
+
+    /// Filter tokens by symbol
+    pub fn symbol(mut self, symbol: String) -> Self {
+        self.filters.insert("symbol", symbol);
+        self
+    }
+
+    /// Filter tokens with `decimals >= min_decimals`
+    /// Clears any exact decimals filter
+    pub fn min_decimals(mut self, decimals: u64) -> Self {
+        self.filters.remove("decimals");
+        self.insert("decimals__gt", decimals.saturating_sub(1));
+        self
+    }
+
+    /// Filter tokens with `decimals <= max_decimals`
+    /// Clears any exact decimals filter
+    pub fn max_decimals(mut self, decimals: u64) -> Self {
+        self.filters.remove("decimals");
+        self.insert("decimals__lt", decimals.saturating_add(1));
+        self
+    }
+
+    /// Filter tokens by exact decimals
+    /// Clears any min or max decimals filters
+    pub fn decimals(mut self, decimals: u64) -> Self {
+        self.clear_decimals();
+        self.insert("decimals", decimals);
+        self
+    }
+
+    /// Specify page limit. If more results than limit are returned, results in
+    /// a paginated response
+    pub fn limit(mut self, limit: u64) -> Self {
+        self.insert("limit", limit);
+        self
+    }
+
+    /// Specify offset in results. Used in pagination, not recommended to be
+    /// specified manually
+    pub fn offset(mut self, offset: u64) -> Self {
+        self.insert("offset", offset);
+        self
+    }
+
+    /// Converts to a URL with query string
+    pub fn to_url(self) -> Url {
+        let mut url = self.client.url().clone();
+        url = Self::url(&url);
+        url.query_pairs_mut().extend_pairs(self.filters.iter());
+        url
+    }
+}
+
+/// token info response
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TokenResponse {
+    /// The token type (ERC20, ERC721, etc)
+    #[serde(rename(deserialize = "type"))]
+    // #[serde(skip)]
+    pub token_type: String,
+    /// The address of the token
+    pub address: String,
+    /// The name of the token
+    pub name: String,
+    /// The symbol of the token
+    pub symbol: String,
+    /// The number of decimals of the token
+    pub decimals: Option<u32>,
+    /// The Logo URI of the token, if it exists
+    #[serde(rename(deserialize = "logoUri"))]
```

Comment:
does the `rename_all = "camelCase"` not cover this? 🤔

Reply:
yup it does—fixed in https://github.com/nomad-xyz/safe-sdk/pull/2/commits/bb831908dfadb1b42df5e357b3401ff1fd9ea0ea